OmniKit/MessageTransport/MessageBlocks/PodInfoFaultEvent.swift

```diff
@@ -10,118 +10,133 @@ import Foundation
+        case none                                                     = 0b0000
+        case immediateBolusInProgress                                 = 0b0001
+        case internal2BitVariableSetAndManipulatedInMainLoopRoutines2 = 0b0010
+        case internal2BitVariableSetAndManipulatedInMainLoopRoutines3 = 0b0100
+        case insulinStateTableCorruption                              = 0b1000
+        
+        public var description: String {
+            switch self {
+                case .none:
+                    return "None"
+                case .immediateBolusInProgress:
+                    return "Immediate Bolus In Progress"
+                case .internal2BitVariableSetAndManipulatedInMainLoopRoutines2:
+                    return "Internal 2-Bit Variable Set And Manipulated In Main Loop Routines 0x02"
+                case .internal2BitVariableSetAndManipulatedInMainLoopRoutines3:
+                    return "Internal 2-Bit Variable Set And Manipulated In Main Loop Routines 0x03"
+                case .insulinStateTableCorruption:
+                    return "Insulin State Table Corruption"
+            }
+        }        
+    }
+    
     public init(encodedData: Data) throws {
         
-        if encodedData.count < 19 {
+        if encodedData.count < 21 {
             throw MessageBlockError.notEnoughData
         }
         
-        guard let reservoirStatus = ReservoirStatus(rawValue: encodedData[1]) else {
-            throw MessageError.unknownValue(value: encodedData[1], typeDescription: "ReservoirStatus")
+        guard PodProgressStatus(rawValue: encodedData[1]) != nil else {
+            throw MessageError.unknownValue(value: encodedData[1], typeDescription: "PodProgressStatus")
         }
-        self.reservoirStatus = reservoirStatus
+        self.podProgressStatus = PodProgressStatus(rawValue: encodedData[1])!
         
         self.deliveryType = DeliveryType(rawValue: encodedData[2] & 0xf)
         
         self.insulinNotDelivered = podPulseSize * Double((Int(encodedData[3] & 0x3) << 8) | Int(encodedData[4]))
         
         self.podMessageCounter = encodedData[5]
-        self.unknownPageCode = Double(Int(encodedData[6]) | Int(encodedData[7]))
+        self.unknownPageCode = encodedData[6...7]
         
-        self.originalLoggedFaultEvent = FaultEventCode(rawValue: encodedData[8])
+        self.currentStatus = FaultEventCode(rawValue: encodedData[8])
         
-        self.faultEventTimeSinceActivation = TimeInterval(minutes: Double((Int(encodedData[9] & 0b1) << 8) + Int(encodedData[10])))
+        self.faultEventTimeSinceActivation = TimeInterval(minutes: Double(encodedData[9...10].toBigEndian(UInt16.self)))
         
-        self.insulinRemaining = podPulseSize * Double((Int(encodedData[11] & 0x3) << 8) | Int(encodedData[12]))
+        let resHighBits = Int(encodedData[11] & 0x03) << 6
+        let resLowBits = Int(encodedData[12] >> 2)
```

Comment:
Did we determine that these 2 bits do not have valid info?  I meant to look into this.

Reply:
These 2 bits do return the insulinRemaining. I've tested this with a Pod failure where there was 0.05 units not delivered:
`0216020f0000010200ec6a026803ff026b000028a70820230169` 
I will add this as a Unit test.